### PR TITLE
Permissions issues2

### PR DESCRIPTION
**Pull request by Paul-Ferrell**

### CODE REVIEW

**Comment by Paul-Ferrell** [comment]
- Executable permissions for extracted zips fixed
 - General permissions bugs addressed
 - The general pav.log is now a single file for simplicity.
 - Greatly expanded permission unit tests (mostly by making them very strict).
 - Refactored the extraneous prints test.

File: lib/pavilion/scriptcomposer.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 from pavilion import module_wrapper
 from pavilion.module_actions import ModuleAction
 
+from pavilion.permissions import PermissionsManager
```
Comment (nicholas-sly) [inline lib/pavilion/scriptcomposer.py:12]: This doesn't appear to be used anywhere in this file...

File: lib/pavilion/test_run.py [hunk 2]
```diff
@@ -574,14 +576,18 @@ def get_permissions(pav_cfg, config) -> (str, int):
...
-        umask = config.get('umask', pav_cfg['umask'])
+        umask = config.get('umask')
+        if umask is None:
+            umask = pav_cfg['umask']
         if umask is not None:
             try:
                 umask = int(umask, 8)
             except ValueError:
                 raise RuntimeError(
                     "Invalid umask. This should have been enforced by the "
```
Comment (nicholas-sly) [inline lib/pavilion/test_run.py:587]: Not part of your changes, but this is a little repetitive.

**Review by nicholas-sly — CHANGES_REQUESTED** [review]
Noticed a couple of things.  Otherwise, looks good.

File: lib/pavilion/test_run.py (hunk 2 above)
Comment (Paul-Ferrell) [reply lib/pavilion/test_run.py:587]: It's not. config['umask'] can have a value of `None` in that dictionary. It was a bug I ran across.

**Pull request merged**